code/modules/mob/living/carbon/human/species_types/ghost.dm

```diff
@@ -0,0 +1,182 @@
+		to keep it private.",
+	)
+
+/datum/species/ghost/create_pref_unique_perks()
+	var/list/to_add = list()
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_POSITIVE_PERK,
+		SPECIES_PERK_ICON = "ghost",
+		SPECIES_PERK_NAME = "Incorporeal",
+		SPECIES_PERK_DESC = "Ghost are able to control their body to the extent where you can willingly make yourself able \
+			to phase through anything, including your own equipment.",
+	))
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+		SPECIES_PERK_ICON = "shoe-prints",
+		SPECIES_PERK_NAME = "No Feet",
+		SPECIES_PERK_DESC = "You lack feet, therefore the ability to wear any shoes!",
+	))
+
+	return to_add
+
+/**
+ * Passthrough ability
+ *
+ * Ghost innate ability that allows them to enter ghost mode,
+ * which gives them the phasing ability, but makes them drop everything
+ * and completely unable to wear anything.
+ */
+/datum/action/innate/toggle_passthrough
+	name = "Toggle passthrough"
+	desc = "Toggles your ability to phase through everything, including your gear and any incompatible organs/limbs."
+	button_icon = 'icons/hud/actions.dmi'
+	button_icon_state = "ghost"
+
+/datum/action/innate/toggle_passthrough/Grant(mob/grant_to)
+	. = ..()
+	RegisterSignal(grant_to, COMSIG_CARBON_POST_ATTACH_LIMB, PROC_REF(on_new_limb))
+
+/datum/action/innate/toggle_passthrough/Remove(mob/remove_from)
+	swap_mode(force_off = TRUE)
+	UnregisterSignal(remove_from, COMSIG_CARBON_POST_ATTACH_LIMB)
+	return ..()
+
+/datum/action/innate/toggle_passthrough/Activate()
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	swap_mode()
+
+/datum/action/innate/toggle_passthrough/IsAvailable(feedback)
+	if(LAZYLEN(owner.reagents))
+		if(locate(/datum/reagent/water/holywater) in owner.reagents)
+			return FALSE
+	return ..()
+
+///Called when the owner of this action gets a new limb, if it isn't a ghost-limb
+///the action will turn itself off, and you'll lose said limb if you try using this action again.
+/datum/action/innate/toggle_passthrough/proc/on_new_limb(mob/source, obj/item/bodypart/new_part, special)
+	SIGNAL_HANDLER
+	if(!iscarbon(owner))
+		return
+	if(!(new_part.bodytype & BODYTYPE_GHOST))
+		swap_mode(force_off = TRUE)
+
+///Swaps the mode, allowing us to phase through stuff but drops everything. Optional 'force_off' arg to prevent being able to turn it on.
+/datum/action/innate/toggle_passthrough/proc/swap_mode(force_off)
+	//we can only turn off, early return if we're trying to turn it on instead.
+	if(force_off && HAS_TRAIT(owner, TRAIT_NO_FLOATING_ANIM))
+		return
+
+	var/mob/living/carbon/carbon_owner = owner
+	var/datum/species/carbon_species = carbon_owner.dna.species
+
+	//drop any limbs & organs that can't phase, now that you're phasing.
+	for(var/obj/item/bodypart/bodypart as anything in carbon_owner.bodyparts)
+		if(!(bodypart.bodytype & BODYTYPE_GHOST))
+			bodypart.drop_limb(special = FALSE, dismembered = FALSE, move_to_floor = TRUE)
```

Comment:
Obligatory "if a ghost's chest is bioscrambled / augmented does this gib them"

Reply:
It doesn't drop chests but I'll make the ability unavailable to be consistent.